From 40bc83e899ddc5e231f623d970ebbfcfae4e2750 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 12 May 2020 23:54:02 +0300
Subject: [PATCH] [SW-0] ppboot/preboot: replace preboot_start()

The preboot_start() function is the same as the more generically called
image_start() function. Reuse the later also for Preboot start.

As a side effect got Goya2 ARC, this brings into Ppboot the
image_start() function which has an implementation instead of the
preboot_start() function which had no implementation for that platform.
Since on the Goya2 ARC the call to preboot_start() (and thus for
image_start()) is not reached (because we return to BootROM), this
should not be a problem.

Change-Id: Iad8ab57f942c21386621edbe4777f12c0a64e15d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index 0fccf6a..fea6c37 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -174,5 +174,5 @@
 	preboot_delay();
 	loadaddr = copy_preboot();
 
-	preboot_start(loadaddr);
+	image_start(loadaddr);
 }
diff --git a/common/cmd_module.h b/common/cmd_module.h
index 680f10e..5d787d6 100644
--- a/common/cmd_module.h
+++ b/common/cmd_module.h
@@ -79,7 +79,6 @@
 void module_lkd_disable(void);
 void module_lkd_run(void);
 void module_lkd_run_noret(void);
-void image_start(u64 loadaddr);
 
 /* UART */
 enum cmd_check_res module_uart_check(void);
diff --git a/gaudi/common.c b/gaudi/common.c
index 7ef6d12..aa3d0ea 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -223,13 +223,6 @@
 	image_entry();
 }
 
-void preboot_start(u64 loadaddr)
-{
-	void (*preboot_start)(void) = (void *)loadaddr;
-
-	preboot_start();
-}
-
 void fit_copy_image(u64 dst, const void *buf, int len)
 {
 	preboot_puts_hex_raw("FIT loadaddr: ", (u32)(dst >> 32), false);
diff --git a/goya/common.c b/goya/common.c
index 22f7bfd..77aafb6 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -175,13 +175,6 @@
 	image_entry();
 }
 
-void preboot_start(u64 loadaddr)
-{
-	void (*preboot_start)(void) = (void *)loadaddr;
-
-	preboot_start();
-}
-
 void fit_copy_image(u64 dst, const void *buf, int len)
 {
 	preboot_puts_hex_raw("FIT loadaddr: ", (u32)(dst >> 32), false);
diff --git a/goya2/common.c b/goya2/common.c
index 6ed079b..ebacbad 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -232,13 +232,6 @@
 	image_entry();
 }
 
-void preboot_start(u64 loadaddr)
-{
-	void (*preboot_start)(void) = (void *)loadaddr;
-
-	preboot_start();
-}
-
 void fit_copy_image(u64 dst, const void *buf, int len)
 {
 	preboot_puts_hex_raw("FIT loadaddr: ", (u32)(dst >> 32), false);
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index 545cc46..db80820 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -308,8 +308,3 @@
 {
 	/* Not implemented - ARC BootROM copies and validates preboot */
 }
-
-void preboot_start(u64 loadaddr)
-{
-	/* Not implemented - ARC BootROM validates and starts preboot */
-}
diff --git a/include/preboot.h b/include/preboot.h
index e05b4f4..ee354fb 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -94,7 +94,7 @@
 void preboot_sync_btl(void);
 
 void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name);
-void preboot_start(u64 loadaddr);
+void image_start(u64 loadaddr);
 
 #endif /* __ASSEMBLY__ */
 
